gnovm/stdlibs/testing/fuzzing/mutator.gno

```diff
@@ -0,0 +1,3204 @@
+	}
+}
+
+// randomXXXFrom gradually mutates XXX and return XXX
+func randomUintFrom(u interface{}) interface{} {
+	p := OncePCG().Float64()
+
+	switch v := u.(type) {
+	case uint:
+		if v == 0 {
+			return uint(RandUint64())
+		}
+		var u interface{}
+		switch {
+		case p < 0.3:
+			min := v / 256
+			max := v * 2
+			u = uint(UintRandRange(uint64(min), uint64(max)))
+		case p < 0.6:
+			min := v / 9096
+			max := v * 8
+			u = uint(UintRandRange(uint64(min), uint64(max)))
+		default:
+			u = uint(OncePCG().Uint64())
+		}
+		return u
+
+	case uint8:
+		if v == 0 {
+			return uint8(RandUint64())
+		}
+		var u8 interface{}
+		switch {
+		case p < 0.3:
+			min := int64(v) / 8
+			max := int64(v) * 2
+			u8 = uint8(RandRange(min, max))
+		case p < 0.6:
+			min := int64(v) / 32
+			max := int64(v) * 8
+			u8 = uint8(RandRange(min, max))
+		default:
+			u8 = uint8(RandRange(0, 256))
+		}
+		return u8
+
+	case uint16:
+		if v == 0 {
+			return uint16(RandUint64())
+		}
+		var u16 interface{}
+		switch {
+		case p < 0.3:
+			min := int64(v) / 256
+			max := int64(v) * 2
+			u16 = uint16(RandRange(min, max))
+		case p < 0.6:
+			min := int64(v) / 9096
+			max := int64(v) * 8
+			u16 = uint16(RandRange(min, max))
+		default:
+			u16 = uint16(RandRange(0, 65536))
+		}
+		return u16
+
+	case uint32:
+		if v == 0 {
+			return uint32(RandUint64())
+		}
+		var u32 interface{}
+		switch {
+		case p < 0.2:
+			min := int64(v) / 256
+			max := int64(v) * 2
+			u32 = uint32(RandRange(min, max))
+		case p < 0.4:
+			min := int64(v) / 9096
+			max := int64(v) * 8
+			u32 = uint32(RandRange(min, max))
+		case p < 0.6:
+			min := int64(v) / (9096 * 9096)
+			max := int64(v) * 16
+			u32 = uint32(RandRange(min, max))
+		default:
+			u32 = uint32(RandRange(0, int64(^uint32(0))))
+		}
+		return u32
+
+	case uint64:
+		if v == 0 {
+			return RandUint64()
+		}
+		var u64 interface{}
+		switch {
+		case p < 0.2:
+			min := v / 256
+			max := v * 2
+			u64 = UintRandRange(min, max)
+		case p < 0.5:
+			min := v / 9096
+			max := v * 8
+			u64 = UintRandRange(min, max)
+		case p < 0.5:
+			min := v / (9096 * 9096)
+			max := v * 16
+			u64 = UintRandRange(min, max)
+		default:
+			u64 = RandUint64()
+		}
+		return u64
+
+	default:
+		panic("it's not a supported uint type")
+	}
+}
+
+// randomXXXFrom gradually mutates XXX and return XXX
+func randomFloatFrom(f interface{}) interface{} {
+	switch v := f.(type) {
+	case float32:
+		var f32 float32
+		f32 = randFloat32From(float32(v))
+		return float32(f32)
+	case float64:
+		var f64 float64
+		f64 = randFloat64From(float64(v))
+		return float64(f64)
+	default:
+		panic("argument is not float 32 or float4")
+	}
+}
+
+func randFloat32From(f float32) float32 {
+	return randomFloat32(f)
+}
+
+func randFloat64From(f float64) float64 {
+	return randomFloat64(f)
+}
+
+func randomBool() bool {
+	return UniformRandomBool(0.5)
+}
+
+func randomFloat32(a float32) float32 {
+	bits := math.Float32bits(a)
+
+	exponent := (bits >> 23) & 0xFF
```

Comment:
You can reduce unnecessary type conversions by setting `exponent` to `int32` type. In this case `shift` should also be changed to `int32` type.